test/new-e2e/tests/fips-compliance/fips_ciphers_nix_test.go

```
@@ -0,0 +1,180 @@
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"github.com/pulumi/pulumi/sdk/v3/go/pulumi"
+)
+
+var (
+	testcases = map[string]bool{
+		"ecc -c TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256":       true,
+		"rsa -c TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256":         true,
+		"ecc -c TLS_ECDHE_ECDSA_WITH_AES_256_GCM_SHA384":       true,
+		"rsa -c TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384":         true,
+		"rsa -c TLS_AES_128_GCM_SHA256 --tls-max=1.3":          true,
+		"rsa -c TLS_AES_256_GCM_SHA384 --tls-max=1.3":          true,
+		"ecc -c TLS_ECDHE_ECDSA_WITH_CHACHA20_POLY1305_SHA256": false,
+		"rsa -c TLS_ECDHE_RSA_WITH_CHACHA20_POLY1305_SHA256":   false,
+		"ecc -c TLS_ECDHE_ECDSA_WITH_AES_128_CBC_SHA":          false,
+		"rsa -c TLS_ECDHE_RSA_WITH_AES_128_CBC_SHA":            false,
+		"ecc -c TLS_ECDHE_ECDSA_WITH_AES_256_CBC_SHA":          false,
+		"rsa -c TLS_ECDHE_RSA_WITH_AES_256_CBC_SHA":            false,
+		"rsa -c TLS_ECDHE_RSA_WITH_3DES_EDE_CBC_SHA":           false,
+	}
+)
+
+//go:embed fixtures/docker-compose.yaml.tmpl
+var dockerComposeTemplate string
+
+type fipsServerSuite struct {
+	e2e.BaseSuite[environments.DockerHost]
+}
+
+func TestFIPSCiphersSuite(t *testing.T) {
+	e2e.Run(t, &fipsServerSuite{}, e2e.WithProvisioner(awsdocker.Provisioner()))
+}
+
+func (v *fipsServerSuite) TestFIPSCiphersFIPSEnabled() {
+	templateVars := map[string]interface{}{
+		"FIPSEnabled": "1",
+	}
+	dockerCompose := fillTmplConfig(v.T(), dockerComposeTemplate, templateVars)
+
+	v.UpdateEnv(
+		awsdocker.Provisioner(
+			awsdocker.WithAgentOptions(
+				dockeragentparams.WithExtraComposeManifest("fips-server", pulumi.String(dockerCompose)),
+			),
+		),
+	)
+
+	for command, shouldConnect := range testcases {
+		v.T().Run(fmt.Sprintf("FIPS enabled testing '%v' (should connect %v)", command, shouldConnect), func(t *testing.T) {
+
+			// Start the fips-server and waits for it to be ready
+			runFipsServer(v, command)
```

<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**💬 suggestion:** ‏If you always call `stopFipsServer` after `runFipsServer` then you could `defer stopFipsServer(v)` ?

---

You can also use `v.T().Cleanup(...)` if you can ignore the error in the function (I'm not sure what happens if a `Cleanup` function call `t.Fail`...)

---

2b616725733590d87a7c46b756dea1c8be63b523